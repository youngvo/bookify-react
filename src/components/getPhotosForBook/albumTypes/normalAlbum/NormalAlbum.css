.album{
    margin: 12px 20px 10px 0px;
    height: 150px;
    width: 120px;
    background: linear-gradient(to bottom, #FFFFFF 30%, #DDDDDD 90%);
    border-radius: 8px;
    display: inline-block;
    position: relative;
    box-shadow:
    2px 2px 2px rgba(0, 0, 0, .3),
    4px -3px 0 #fff,
    6px -1px 2px rgba(0, 0, 0, .3),
    8px -7px 0 #fff,
    10px -4px 2px rgba(0, 0, 0, .3);
    left: 10px;
}

.album-smugmug{
    margin: 12px 20px 10px 0;
    height: 150px;
    width: 120px;
    background: linear-gradient(to bottom, #FFFFFF 30%, #DDDDDD 90%);
    border-radius: 8px;
    display: inline-block;
    position: relative;
    box-shadow:
    2px 2px 2px rgba(0, 0, 0, .3);
    left: 10px;
}

.album-can-not-click{
    margin: 70px 20px 30px 0px;
    height: 150px;
    width: 120px;
    background: linear-gradient(to bottom, #FFFFFF 30%, #DDDDDD 90%);
    border-radius: 8px;
    display: inline-block;
    position: relative;
    box-shadow:
    2px 2px 2px rgba(0, 0, 0, .3),
    4px -3px 0 #fff,
    6px -1px 2px rgba(0, 0, 0, .3),
    8px -7px 0 #fff,
    10px -4px 2px rgba(0, 0, 0, .3);
    left: 10px;
}

.album-smugmug-can-not-click{
    margin: 12px 20px 30px 0;
    height: 150px;
    width: 120px;
    background: linear-gradient(to bottom, #FFFFFF 30%, #DDDDDD 90%);
    border-radius: 8px;
    display: inline-block;
    position: relative;
    box-shadow:2px 2px 2px rgba(0, 0, 0, .3);
    top: 50px;
    left: 10px;
}



.album-smugmug:hover{
    background:#a9e1fa;
    cursor: pointer;
}

.album:hover {
    background:#a9e1fa ;
    cursor: pointer;
}

.image-preview-album {
    max-height: 100px;
    min-width: 75px;
}

.view-info-album-zone * {
    padding: 0;
    margin: 0px 10px 0 8px;
    color: dimgray;
}

.name-album {
    font-size: 13px;
    font-weight:bold;
    color:#555;
}

.name-album-long-text {
    font-size: 13px;
    font-weight:bold;
    color:#555;
    white-space: nowrap;
    width: 100px;
    overflow: hidden;
    text-overflow: ellipsis;

}


.amount-of-photo {
    font-size: 15px;
    font-weight:bold;
    color:#9c9c9c;
}

.active {
    border-radius: 15px;
    border: 3px solid #EC6E1A;
}

.inner-shadow-box{
    border: 1px hidden;
    height: 147px;
    width: 117px;
    background: transparent;
    border-radius: 5px;
    box-shadow: inset 2px 2px 2px rgba(0, 0, 0, .3)
}

.outer-shadow-box{
    border: 3px solid #EC6E1A;
    height: 147px;
    width: 117px;
    background: transparent;
    border-radius: 8px;
    box-shadow: 2px 2px 2px rgba(0, 0, 0, .3)
}

.badge-notification {
    cursor: default;
    color: white;
    font-size: 15px;
    text-align: center;
    position: absolute;
    min-width: 25px;
    background: #EC6E1A;
    border-radius: 15px;
    border: solid white 2px;
    right: 5px;
    bottom: 0;
    transform: translate(50%, 50%);
    padding: 2px 4px;
}

.preview-album-zone{
    margin-bottom: 7px;
    padding: 10px 0px 0px 20px;
    width: 75px;
    height: 75px;
    overflow: hidden;
}