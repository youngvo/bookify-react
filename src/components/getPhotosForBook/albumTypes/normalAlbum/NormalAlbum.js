import React from 'react';

import './NormalAlbum.css';

const renderCountPhotosSelected = (photoSelectedCount) => {
    if (photoSelectedCount && photoSelectedCount > 0) {
        // return (<div>{photoSelectedCount}</div>);
        return <BadgeNotification photoSelectedCount={photoSelectedCount} />
    }
}

const BadgeNotification = function({photoSelectedCount}) {
    return (
        <div className="badge-notification">{photoSelectedCount}</div>
    );
}

const renderPicture = (picUrl) => (
    <div className="preview-album-zone" >
        {picUrl && <img className="image-preview-album" alt={picUrl} src={picUrl} />}
    </div>
);

const renderInfo = (albumName, photoCount) => (
    <div className="view-info-album-zone">
        <p className="name-album-long-text">{albumName}</p>
        <p className="amount-of-photo">{`${photoCount} photos`}</p>
    </div>
);

const onAlbumClick = (album, onClick) => {
    if (onClick) onClick(album);
}

const NormalAlbum = ({ album, photoSelectedCount, onClick }) => {
    let styleCss = '';
    if(onClick) {
        if (album.type == 'px500') {
            styleCss = 'album';
        } else {
            styleCss = 'album-smugmug';
        }
    } else {
        if (album.type == 'px500') {
            styleCss = 'album-can-not-click';
        } else {
            styleCss = 'album-smugmug-can-not-click';
        }
    }
    
    return (
        <div className={styleCss} 
            onClick={onAlbumClick.bind(this, album, onClick)} >
            <div className={photoSelectedCount > 0 ? "outer-shadow-box" : ""}>
                <div className={photoSelectedCount > 0 ? "inner-shadow-box" : ""}>
                    {renderCountPhotosSelected(photoSelectedCount)}
                    {renderPicture(album.picUrls[0])}
                    {renderInfo(album.name, album.photoCount!=0 ? album.photoCount : album.photoIds.length)}
                </div>
            </div>
        </div>
    )
};

export default NormalAlbum;
