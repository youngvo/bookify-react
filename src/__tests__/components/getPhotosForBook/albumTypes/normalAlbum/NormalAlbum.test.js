import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import NormalAlbum from './../../../../../components/getPhotosForBook/albumTypes/normalAlbum/NormalAlbum';

describe('Test NormalAlbum Component', () => {
    let album = {
        name: 'album',
        photoIds: [],
        picUrls: ["http://airbnb.io/enzyme/docs/api/ShallowWrapper/children.html"],
    };
    let photoSelectedCount = 0; 
    let onClick;
    let normalAlbum;
    beforeEach(() => {
        onClick = jest.fn();
        normalAlbum = shallow(<NormalAlbum album={album} photoSelectedCount={photoSelectedCount} onClick={onClick}/>)
    })

    describe('NormalAlbum Snapshot', () => {
        it('NormalAlbum does not change UI', () => {
            const component = renderer.create(<NormalAlbum album={album} photoSelectedCount={photoSelectedCount} onClick={onClick}/>);
            const tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('NormalAlbum DOM renders', () => {
        
        it('Cotains div with className="album active" if photoSelectedCount > 0 ', () => {
            normalAlbum.setProps({ photoSelectedCount: 10 });
            expect(normalAlbum.find('.album .active').length).toEqual(1);

        }); 

        it('Cotains div with className="album" if photoSelectedCount = 0', () => {
            expect(normalAlbum.find('.album').length).toEqual(1);
        }); 

        describe('test renderCountPhotosSelected() return ', () => {
            it('Contain div with BadgeNotification if photoSelectedCount > 0 ', () => {
                normalAlbum.setProps({ photoSelectedCount: 10 });
                expect(normalAlbum.find('BadgeNotification').length).toEqual(1);
            });

            it(' Not Contain div with BadgeNotification if photoSelectedCount = 0 ', () => {
                normalAlbum.setProps({ photoSelectedCount: 0 });
                expect(normalAlbum.find('BadgeNotification').length).toEqual(0);
            });
        })

        describe('test renderInfo() return ',() => {
            it('renderInfo(), render album name', () => {
                expect(normalAlbum.find('p.name-album').length).toEqual(1);
                expect(normalAlbum.find('p.name-album').text()).toEqual(album.name);
            });
            
            it('renderInfo(), render amount of Photo', () => {
                expect(normalAlbum.find('p.amount-of-photo').length).toEqual(1);
                expect(normalAlbum.find('p.amount-of-photo').text()).toEqual(album.photoIds.length + ' photos');
            });
        });

        describe('test renderPicture() return ', () => {
            it('return div with className="preview-album-zone"', () => {
                expect(normalAlbum.find('.preview-album-zone').length).toEqual(1);
            })
            it('it contains <img/> with className="image-preview-album" tag', () => {
                expect(normalAlbum.find('img.image-preview-album').length).toEqual(1);
            })
        });

    })

    describe('NormalAlbum test funtion', () => {
        it('test onAlbumClick() will be dispatch onClick()', () => {
            normalAlbum.simulate('click');
            //normalAlbum.instance().onAlbumClick(album, onClick);
            expect(onClick.mock.calls.length).toBe(1);
            normalAlbum.simulate('click');
            expect(onClick.mock.calls.length).toBe(2);
        });
    })

});


