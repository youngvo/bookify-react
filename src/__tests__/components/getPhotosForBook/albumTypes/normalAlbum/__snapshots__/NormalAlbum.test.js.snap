exports[`Test NormalAlbum Component NormalAlbum Snapshot NormalAlbum does not change UI 1`] = `
<div
  className="album"
  onClick={[Function]}>
  <div
    className="preview-album-zone">
    <img
      alt="http://airbnb.io/enzyme/docs/api/ShallowWrapper/children.html"
      className="image-preview-album"
      src="http://airbnb.io/enzyme/docs/api/ShallowWrapper/children.html" />
  </div>
  <div
    className="view-info-album-zone">
    <p
      className="name-album">
      album
    </p>
    <p
      className="amount-of-photo">
      0 photos
    </p>
  </div>
</div>
`;
